frontend/src/services/action-items-service.ts

```diff
@@ -0,0 +1,21 @@
+import { AxiosInstance } from 'axios';
+import { createApiClient } from './api-client';
+import { IActionItem, IWeeklyActionItems } from 'shared-types';
+
+export class ActionItemsService {
+  private apiClient: AxiosInstance;
+
+  constructor() {
+    this.apiClient = createApiClient('/action-items');
+  }
+
+  async getWeeklyActionItems(): Promise<IWeeklyActionItems>{
+   return (await this.apiClient.get('')).data;
+  }
+
+  async updateActionItemStatus(item: IActionItem, itemsId: string): Promise<void> {
+    await this.apiClient.put(`/?id=${itemsId}`, item);
```

Comment:
you can send a params in the `config` argument (3rd argument for put) dictionary to avoid concatenating string in a `?key=value&` format.
Try this suggestion
```suggestion
    await this.apiClient.put('/', item, { params: { id: itemsId } });
```